### PR TITLE
07 olegs kozlovs

### PR DESCRIPTION
**Pull request by zoyazip**

### CODE REVIEW

**Review by zoyazip — COMMENTED** [review]
Iesk!
You can try to boost your app, by adding a validation to input field (For example, if task is empty)

File: Majasdarbs/07-majasdarbs/OlegsKozlovs/app.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  for (let i = 0; i < taskList.length; i++) {
+    if (taskList[i].value === elem) {
+      return true;
+    }
+  }
+  return false;
+}
+
+const isValidInput = (input) => {
+  if (input === null || input.trim().length === 0) {
```
Comment (ruback105) [inline Majasdarbs/07-majasdarbs/OlegsKozlovs/app.js]: You can remove this validation and make field required in html https://www.w3schools.com/tags/att_input_required.asp

File: Majasdarbs/07-majasdarbs/OlegsKozlovs/app.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  inputField.value = "";
+  renderTask();
+};
+
+const toggleDone = (e) => {
+  let clickedLiIndex;
+  if (e.srcElement.localName === "li") {
+    // Complete done toggle
+    clickedLiIndex = Array.from(taskListElement.children).indexOf(e.target);
+    if (taskList[clickedLiIndex].done) {
```
Comment (ruback105) [inline Majasdarbs/07-majasdarbs/OlegsKozlovs/app.js]: this if else block can be written in 1 line

taskList[clickedLiIndex].done = !taskList[clickedLiIndex].done

Basically you just use the same value, but reverse it.

File: Majasdarbs/07-majasdarbs/OlegsKozlovs/app.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  saveToLocalStorage("task_list", taskList);
+  inputField.value = "";
+  renderTask();
+};
+
+const toggleDone = (e) => {
+  let clickedLiIndex;
+  if (e.srcElement.localName === "li") {
+    // Complete done toggle
+    clickedLiIndex = Array.from(taskListElement.children).indexOf(e.target);
```
Comment (ruback105) [inline Majasdarbs/07-majasdarbs/OlegsKozlovs/app.js]: One optimisation can be done here - you could add to each li element data-id="" attribute, and then use it to find out, which li item was clicked.

And the id should be equal to element position inside the array, so first item should have id="0", second id="1", etc.

File: Majasdarbs/07-majasdarbs/OlegsKozlovs/app.js (hunk 1 above)
Comment (Olegs-github) [reply Majasdarbs/07-majasdarbs/OlegsKozlovs/app.js]: Thanks for comment. I have tried to do that but for some reason "required" option in HTML doesn't work for me. I have done the following: input type="text" name="task" placeholder="New Task Item" id="input" required (tag open/close marks are skipped here to ensure line showing in comments)
Reviewing potential problems with this option in Google, I can't find the root of the issue. Could you please advise?

**Pull request closed without merging**